internal/x/protoc/downloader.go

```diff
@@ -159,7 +159,13 @@ func (d *downloader) downloadInternal(basePath string, goos string, goarch strin
 		return err
 	}
 	response, err := http.Get(url)
-	if err != nil {
+	if err != nil || response.StatusCode != http.StatusOK {
+		// if there is not given protocURL, we tried to
+		// download this from GitHub Releases, so add
+		// extra context to the error message
+		if d.protocURL == "" {
+			return fmt.Errorf("error downloading %s: %v\nmake sure GitHub Releases has a proper protoc zip file of the form protoc-VERSION-OS-ARCH.zip at https://github.com/google/protobuf/releases/v%s\nnote that many micro versions do not have this, and no version before 3.0.0-beta-2 has this", url, err, d.config.Compile.ProtobufVersion)
```

Comment:
That's fair. To avoid a poorly punctuated error message, we could update to:
```
"error downloading %s: %v\nMake sure GitHub Releases has a proper protoc zip file of the form protoc-VERSION-OS-ARCH.zip at https://github.com/google/protobuf/releases/v%s"
```

This way we continue to be consistent with Golang errors, but use the newline as a separator for a more-informative sentence.

Reply:
Is that just capitalizing the M? What about the rest of the message?

Reply:
Capitalized the sentences - is this it? Do you want the latter part of the error message deleted? I have it in there because it's informative for users trying to use proto2, see #63.

Reply:
I meant to remove the remaining note since I thought it was giving more information than needed. But if you think this will help enough people to justify keeping it, then I'm OK with it :+1: